Makefile

```diff
@@ -38,7 +38,6 @@ test-all:
 	tox
 
 build-docs:
-	sphinx-apidoc -o docs/ . setup.py "*conftest*"
```

Comment:
I considered leaving auto generation of api docs in but I have certain assumptions regarding how docs should look like, what things go in the toctree etc and I just couldn't find a way to have full auto generation work in a way that spits out something of high quality (subjective of course).

I find `autoclass` already to be of an tremendous help and think it's better to put *some* manual labor into the very last mile to arrive at a result that doesn't look super generic and machine generated.